nomad/state/state_store_host_volumes_test.go

```diff
@@ -295,4 +295,24 @@ func TestStateStore_UpdateHostVolumesFromFingerprint(t *testing.T) {
 		must.Sprint("node pool change should update pending volume"))
 	must.Eq(t, "new-node-pool", vol2.NodePool)
 	must.Eq(t, structs.HostVolumeStateReady, vol2.State)
+
+	// node restarts and fails to restore
+	node = node.Copy()
+	node.HostVolumes = map[string]*structs.ClientHostVolumeConfig{
+		"static-vol": {Name: "static-vol", Path: "/srv/static"},
+	}
+	index++
+	must.NoError(t, store.UpsertNode(structs.MsgTypeTestSetup, index, node))
+
+	vol0, err = store.HostVolumeByID(nil, ns, vols[0].ID, false)
+	must.NoError(t, err)
+	must.Eq(t, index, vol0.ModifyIndex,
+		must.Sprint("failed restore should update ready volume"))
+	must.Eq(t, structs.HostVolumeStateUnavailable, vol0.State)
+
+	vol1, err = store.HostVolumeByID(nil, ns, vols[0].ID, false)
```

Comment:
```suggestion
	vol1, err = store.HostVolumeByID(nil, ns, vols[1].ID, false)
```

right?

also maybe add an extra bit at the bottom around line 318 that puts one back, to ensure that it gets set from unavailable->ready

Reply:
Oops, good catch. And agreed.